3 - PB/Documentazione interna/Norme di Progetto/Norme di Progetto.typ

```diff
@@ -1697,43 +1697,81 @@ Gli argomenti principali trattati nel documento sono due:
+Questo vale sia per il codice sviluppato in proprio sia per quello riutilizzato o esterno, che viene adattato per soddisfare i requisiti specifici.
+
+Per aiutare il lavoro di verifica da parte del Verificatore, vengono riportate in Pull Request tutte le funzionalità che si ha necessità di controllare manualmente, oltre al riferimento al task interessato su Jira.
+
+Una volta che i singoli moduli sono stati sviluppati e testati mediante i test di unità, comincia il processo di integrazione (@processo_integrazione), durante il quale vengono assemblati e combinati per formare il sistema completo.
+A questo punto, vengono eseguiti i test di integrazione per verificare che i vari moduli interagiscano correttamente tra loro e che il sistema nel suo complesso funzioni come previsto.
+
+==== Implementazione
+
+Decisa una strategia di testing per l'elemento di sistema e scritti i relativi test di unità, il Programmatore può quindi procedere allo sviluppo o all'adeguamento del software.
+
+Per garantire una standardizzazione e una migliore gestione del codice, devono essere adottate le seguenti norme e pratiche di organizzazione:
+
++ ciascuna categoria di elementi deve essere assegnata a una cartella specifica (@repository-github):
+
+    - tutti i file correlati ad uno specifico pattern devono essere organizzati all'interno di una cartella dedicata esclusivamente a quel pattern;
+    - tutti i file riguardanti l'interfaccia utente devono essere raggruppati in una cartella apposita;
+    - se all'interno di una cartella sono presenti più file legati ad una stessa classe del modello, devono essere raggruppati in una sottocartella dedicata.
+
++ ogni componente e ogni classe deve essere implementata in un proprio file. Questa pratica semplifica la gestione e la ricerca dei componenti all'interno del progetto, garantendo una maggiore chiarezza e coerenza nel codice. Può essere fatta un'eccezione a questa norma nel caso in cui vengano adottati design pattern che risultano più manutenibili se le loro classi sono implementate tutte nello stesso file;
+
++ i nomi di variabili, metodi e funzioni devono essere parlanti, ovvero devono riflettere il loro scopo e la loro funzione all'interno del sistema, così da facilitare la comprensione e manutenzione del codice;
+
++ regole di codifica:
+
+  - type safety con Typescript attraverso l'utilizzo di tipi specifici e limitando al massimo l'uso del tipo `any`;
+  - notazione camel per nomi di file e variabili;
+  - indentazione del codice di tipo Egyptian braces;
+  - limitare il più possibile la lunghezza di metodi e delle singole righe di codice, perseguendo i principi di leggibilità e manutenibilità.
```

Comment:
In questo punto bisognerebbe indicare il valore numerico sia di lunghezza che di larghezza dei metodi, ma al momento non lo so dire nemmeno io... lascia un TODO ma imposta la frase per accogliere questi valori numerici tipo "Lunghezza massima dei metodi \#TODO"